lesson_25/sources/vector3d.cpp

```diff
@@ -0,0 +1,94 @@
+#include <cmath>
+
+#include "vector3d.hpp"
+
+
+vector3d::vector3d() : vector3d(0, 0, 0)
+{
+}
+
+vector3d::vector3d(const float x, const float y, const float z) : vector3d(0, 0, 0, x, y, z)
```

Comment:
what's wrong with direct assignment:
x_(x), y_(y), z_(z) ?

Reply:
I guess nothing wrong in this case. But for extension purpose can be some issue there. For instance we had a task about static instances counter and it's nice to have one constructor which is base for all others. So in general it can be just a good habit to have one focal point for instantiation and in a future there can be add some logic or maybe dev/debug information etc. but as for me it's really depends.